syncstorage/src/tokenserver/db/models.rs

```diff
@@ -164,7 +164,7 @@ impl TokenserverDb {
              WHERE service = ?
                AND email = ?
                AND generation <= ?
-               AND COALESCE(keys_changed_at, 0) <= COALESCE(?, 0)
+               AND COALESCE(keys_changed_at, 0) <= COALESCE(?, keys_changed_at, 0)
```

Comment:
I had left this bit out the query, but the Python Tokenserver has it: https://github.com/mozilla-services/tokenserver/blob/master/tokenserver/assignment/sqlnode/sql.py#L83

I'm not entirely sure what its purpose is, but it seems like these queries should match

Reply:
Ah, ok.
So, check that the value of `keys_changed_at` or `0` is greater than the passed value, `keys_changed_at`, or `0` . 

Yeah, I can see where that would be good. 

If, for whatever reason, the value you're passing is `null` you could with up with `keys_changed_at <= 0` rather than `keys_changed_at <= keys_changed_at`

Reply:
Ah! I see